src/App.js

```diff
@@ -1,7 +1,85 @@
+import { useState, useEffect } from 'react'
+import MyMap from './components/MyMap'
+import UserLocationText from './components/UserLocationText'
+import CoffeeShopsText from './components/CoffeeShopsText'
+
 function App() {
+  const [userLocationState, setUserLocationState] = useState({
+    latitude: null,
+    longitude: null,
+    isLoading: false,
+    error: null
+  })
+
+  const [coffeeShopsState, setCoffeeShopsState] = useState({
+    isLoading: false,
+    coffeeShops: null,
+    fetchError: null
+  })
```

Comment:
How about an enum like structure to hold the state information, it would make checking for state a bit more concise maybe.

`const DataState = {
    LOADING: 0,
    ERROR: 1,
    AVAILABLE: 2
}`

Reply:
This sounds like a good idea. :+1: